toxcore/os_random.h

```diff
@@ -0,0 +1,26 @@
+/* SPDX-License-Identifier: GPL-3.0-or-later
+ * Copyright © 2022-2025 The TokTok team.
+ */
+
+#ifndef C_TOXCORE_TOXCORE_OS_RANDOM_H
+#define C_TOXCORE_TOXCORE_OS_RANDOM_H
+
+#include "tox_random.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+extern const Tox_Random os_random_obj;
+
+/** @brief System random number generator.
+ *
+ * Uses libsodium's CSPRNG (on Linux, `/dev/urandom`).
```

Comment:
I am not sure which platforms actually use the CSPRNG, since it is only used as a fallback.

Linux uses the getrandom syscall, fallsback to /dev/random (or /dev/urandom)
Windows uses the RtlGenRandom syscall, which we dont know the details off, but on wine this falls back to /dev/random (or 0 if those devices dont exist) and on reactos they use a LCG with current time, which they even say is absolutely not usable for crypto.
Not sure about other platform.

Reply:
getrandom:
> These bytes can be used to seed user-space random number generators or for cryptographic purposes.

If it can be used for cryptographic purposes, I think it can be considered a CSPRNG.

/dev/random:
> In Unix-like operating systems, /dev/random and /dev/urandom are special files that serve as cryptographically secure pseudorandom number generators (CSPRNGs).

Windows: ack.

Ultimately though, for this comment it doesn't matter, because libsodium claims it's a CSPRNG, so it is. If it's not, it's a libsodium bug and you should file one. From libsodium docs:
> The library provides a set of functions to generate unpredictable data, suitable for creating secret keys.

Reply:
I guess unless you consider "[a function] [...] suitable for creating secret keys" not to be cryptographically secure, in which case I'd like to know what you do consider such.

Reply:
> I guess unless you consider "[a function] [...] suitable for creating secret keys" not to be cryptographically secure, in which case I'd like to know what you do consider such.

Right. Did not think this part all the way through.